pkg/ipamd/datastore/data_store.go

```diff
@@ -771,11 +788,6 @@ func (ds *DataStore) getDeletableENI(warmIPTarget, minimumIPTarget, warmPrefixTa
 			continue
 		}
 
-		if ds.isPDEnabled && warmPrefixTarget != 0 && ds.isRequiredForWarmPrefixTarget(warmPrefixTarget, eni) {
```

Comment:
Is it safe to remove this check if prefixes are spread across multiple ENIs? Can we check that flow to be sure?

Reply:
Yeah, this is needed since if a ENI has more than one prefix we should not be deleting the ENI even if the caller thinks ENI is deletable. Somehow this creeped in :)